Sources/MapboxMaps/Location/InterpolatedLocationProducer.swift

```diff
@@ -15,10 +18,19 @@ internal final class InterpolatedLocationProducer: NSObject, InterpolatedLocatio
     private let locationInterpolator: LocationInterpolatorProtocol
     private let dateProvider: DateProvider
 
+    private let consumers = NSHashTable<PuckLocationConsumer>.weakObjects()
+    private var cancelableToken: Cancelable?
```

Comment:
The token can be cancelled on deinit so the observable interpolated location won't notify already deallocated producer.

Reply:
sorry, I don't get what you are suggesting

Reply:
Something like:
```swift
deinit {
    cancelableToken?.cancel()
}
```

Reply:
I'm afraid it might be a significant change for a customers as most of the requests would fail eventually if there is no strong reference on cancellation token.

Reply:
aha, that's right :-D